src/AbstractDependencyRewriter.php

```diff
@@ -0,0 +1,156 @@
+    protected $io;
+
+    /**
+     * @var Replacements
+     */
+    private $replacements;
+
+    public function __construct()
+    {
+        $this->replacements = new Replacements();
+    }
+
+    public function activate(Composer $composer, IOInterface $io)
+    {
+        $this->composer = $composer;
+        $this->io = $io;
+        $this->output(sprintf('<info>Activating %s</info>', get_class($this)), IOInterface::DEBUG);
+    }
+
+    /**
+     * When a ZF package is requested, replace with the Laminas variant.
+     *
+     * When a `require` operation is requested, and a ZF package is detected,
+     * this listener will replace the argument with the equivalent Laminas
+     * package. This ensures that the `composer.json` file is written to
+     * reflect the package installed.
+     */
+    public function onPreCommandRun(PreCommandRunEvent $event)
+    {
+        $this->output(
+            sprintf(
+                '<info>In %s::%s</info>',
+                get_class($this),
+                __FUNCTION__
+            ),
+            IOInterface::DEBUG
+        );
+
+        if (! in_array($event->getCommand(), ['require', 'update'], true)) {
+            // Nothing to do here.
+            return;
+        }
+
+        $input = $event->getInput();
+        if (! $input->hasArgument('packages')) {
+            return;
+        }
+
+        $input->setArgument(
+            'packages',
+            array_map([$this, 'updatePackageArgument'], $input->getArgument('packages'))
+        );
+    }
+
+    abstract public function onPrePackageInstallOrUpdate(PackageEvent $event);
+
+    /**
+     * @param string $message
+     * @param int $verbosity
+     */
+    protected function output($message, $verbosity = IOInterface::NORMAL)
+    {
+        $this->io->write($message, $newline = true, $verbosity);
+    }
+
+    /**
+     * Parses a package argument from the command line, replacing it with the
+     * Laminas variant if it exists.
+     *
+     * @param string $package Package specification from command line
+     * @return string Modified package specification containing Laminas
+     *     substitution, or original if no changes required.
+     */
+    private function updatePackageArgument($package)
+    {
+        $result = preg_split('/[ :=]/', $package, 2);
+        if ($result === false) {
+            return $package;
+        }
+        $name = array_shift($result);
+
+        if (! $this->isZendPackage($name)) {
+            return $package;
+        }
+
+        $replacementName = $this->transformPackageName($name);
+        if ($replacementName === $name) {
+            return $package;
+        }
+
+        $this->io->write(
+            sprintf(
+                'Changing package in current command from %s to %s',
+                $name,
+                $replacementName
+            ),
+            true,
+            IOInterface::DEBUG
+        );
+
+        $version = count($result) ? array_shift($result) : null;
```

Comment:
no need count, can use truthy check:

```suggestion
        $version = $result ? array_shift($result) : null;
```

Reply:
Same here, this is from current behavior. Just copy & pasted to new class